### DISCUSSION TITLE
Question about getConfigByChainId Function in Lesson 20 of Cyfrin Updraft(https://updraft.cyfrin.io/courses/foundry/smart-contract-lottery/deploy-script?lesson_format=video)

### DISCUSSION THREAD

**Discussion by meitedaf** [Cyfrin/foundry-full-course-cu#2035] [Q&A] [answered]
Hi everyone,

I’m wondering about the implementation of the getConfigByChainId function in Lesson 20. My question is why there is no need for judgment on Sepolia like this:

```
function getConfigByChainId(
    uint256 chainId
) public returns (NetworkConfig memory) {
    if (networkConfigs[chainId].vrfCoordinator != address(0)) {
        return networkConfigs[chainId];
    } else if (chainId == SEPOLIA_CHAIN_ID) {
        return getSepoliaEthConfig();
    } else if (chainId == DEFAULT_ANVIL_ID) {
        return getOrCreateAnvilEthConfig();
    } else revert HelperConfig__InvalidChainId();
}
```

In the video, the function is:
```
function getConfigByChainId(uint256 chainId) public returns (NetworkConfig memory) {
    if (networkConfigs[chainId].vrfCoordinator != address(0)) {
        return networkConfigs[chainId];
    } else if (chainId == LOCAL_CHAIN_ID) {
        return getOrCreateAnvilEthConfig();
    } else {
        revert HelperConfig__InvalidChainId();
    }
}
```

Why does the video implementation not include a specific judgment for Sepolia like in the first example? Any insights would be greatly appreciated!

Thanks in advance.

**Comment by EngrPips** [comment, marked as answer]
Hello @meitedaf, The reason why there is not a conditional check when we are getting the `NetworkConfig` on sepolia is because every detail we need on sepolia to fill the `NetworkConfig` is always readily available, but for the `Anvil` chain, we need to always deploy some contract the first time we call the `HelperConfig.` Examples of such contracts are `VRFCoordinator` and `Link Token.` Please ask questions if anything is confusing.

**Reply by meitedaf** [reply]
Thank you so much for the timely help! I understand now.

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it.
